app/lib/efile/nj/nj_retirement_income_helper.rb

```diff
@@ -45,20 +69,38 @@ def retirement_exclusion_eligible?(line_15, line_27, line_28a)
         end
         return false if line_15 > 3_000
         return false if line_27 > 150_000
-        return false if line_28a > calculate_maximum_exclusion(line_27)
+        return false if line_28a > calculate_maximum_exclusion(line_27, line_27)
 
         true
       end
 
-      def calculate_maximum_exclusion(total_income)
+      def line_28a_eligible?(line_27)
+        return false if line_27 > 150_000
+        if @intake.spouse_birth_date.present?
+          return false unless line_28a_primary_eligible? || line_28a_spouse_eligible?
+        else
+          return false unless line_28a_primary_eligible?
+        end
+        true
+      end
+
+      def line_28a_primary_eligible?
+        primary_62_and_older? || @intake.primary_disabled_yes? || @intake.direct_file_data.is_primary_blind?
+      end
+
+      def line_28a_spouse_eligible?
+        spouse_62_and_older? || @intake.spouse_disabled_yes? || @intake.direct_file_data.is_spouse_blind?
+      end
```

Comment:
Compared to the 28b pattern, pulling out these methods prevents duplicative logic between the `eligible?` function and the total_eligible_income function. 

John and I discussed a fast-follow ticket where we would refactor 28b to use this pattern as well.